chore(docs): update info on router versions (#23075)
expo/expo: docs/pages/router/reference/typed-routes.mdx

import { Tabs, Tab } from '~/ui/components/Tabs';
import { FileTree } from '~/ui/components/FileTree';

> > Available from Expo SDK 49 and Expo Router v2. Expo Router supports standard TypeScript out of the box. See [TypeScript](/guides/typescript/) guide for more information on how to set it up.

Author: ```suggestion
> Available from Expo SDK 49 and Expo Router v2. Expo Router supports standard TypeScript out of the box. See the [TypeScript](/guides/typescript/) guide for more information on how to set it up.
```

> > First available in Expo SDK 49 and Expo Router v2. Expo Router supports standard TypeScript out of the box. See [TypeScript](/guides/typescript/) guide for more information on how to set it up.

Reviewer: ```suggestion
> Available from Expo SDK 49 and Expo Router v2. Expo Router supports standard TypeScript out of the box. See [TypeScript](/guides/typescript/) guide for more information on how to set it up.
```
